test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewComponents/DefaultViewComponentSelectorTest.cs

```diff
@@ -112,6 +112,26 @@ public void SelectComponent_Ambiguity()
             Assert.Equal(expected, ex.Message);
         }
 
+        [Theory]
+        [InlineData("Name")]
+        [InlineData("Ambiguous.Name")]
+        public void SelectComponent_AmbiguityDueToDerivation(string name)
+        {
+            var selector = CreateSelector();
+            var expected =
+                $"The view component name '{name}' matched multiple types:" + Environment.NewLine +
+                $"Type: '{typeof(ViewComponentContainer.AmbiguousBase)}' - " +
+                "Name: 'Ambiguous.Name'" + Environment.NewLine +
+                $"Type: '{typeof(ViewComponentContainer.DerivedAmbiguous)}' - " +
```

Comment:
AppVeyor, Travis, and local tests are working fine. But I agree this and one of the existing tests assume an invariant that doesn't exist. I made a change later in the class to order types alphabetically.

FYI everything comes down to the `Assembly.DefinedTypes` order in real life i.e. when using `AssemblyPart`. `Type.GetNestedTypes()` controls the order in this test class. Either way, the order isn't defined and the [`GetNestedTypes()` docs](https://msdn.microsoft.com/en-us/library/493t6h7t(v=vs.110).aspx#Anchor_1) underscore that point.